mne_bids/write.py

```diff
@@ -952,3 +950,124 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
+
+    Parameters
+    ----------
+    bids_root : str
+        Path to root of the BIDS folder
+    subject : str
+        Subject label as in 'sub-<label>', for example: '01'
+    t1w : str
+        Path to a T1 weighted MRI scan of the subject. Must be in .nii,
+        .nii.gz, or .mgz format. If in .mgz, it will be converted to .nii.gz
+    session : str | None
+        The session for `t1w`. Corresponds to "ses"
+    acquisition: str | None
+        The acquisition parameters for `t1w`. Corresponds to "acq"
+    raw : instance of Raw | None
+        The raw data of `subject` corresponding to `t1w`. If `raw` is None,
+        `trans` has to be None as well
+    trans : instance of mne.transforms.Transform | None
+        The transformation matrix from HEAD coordinates to MRI coordinates.
+        If None, no sidecar JSON file will be written for `t1w`
+    overwrite : bool
+        Whether to overwrite existing files or data in files.
+        Defaults to False.
+        If overwrite is True, any existing files with the same BIDS parameters
+        will be overwritten with the exception of the `participants.tsv` and
+        `scans.tsv` files. For these files, parts of pre-existing data that
+        match the current data will be replaced.
+        If overwrite is False, no existing data will be overwritten or
+        replaced.
+    verbose : bool
+        If verbose is True, this will print a snippet of the sidecar files. If
+        False, no content will be printed.
+
+    Returns
+    -------
+    anat_dir : str
+        Path to the anatomical scan in the `bids_dir`
+
+    """
+    # Make directory for anatomical data
+    anat_dir = op.join(bids_root, 'sub-{}'.format(subject),
+                       'ses-{}'.format(session), 'anat')
+    if not op.exists(anat_dir):
+        os.makedirs(anat_dir)
+    else:
+        raise IOError('An /anat directory for sub-{} already exists in {}'
+                      .format(subject, bids_root))
+
+    # Check which type ot T1 MRI data we have. For now, we will need NIfTI for
+    # saving as BIDS ... and .mgz for extracting transformation matrices
+    # XXX: https://neurostars.org/t/get-voxel-to-ras-transformation-from-nifti-file/4549  # noqa: E501
+    _, ext = _parse_ext(t1w)
+    if ext not in ['.nii', '.nii.gz', '.mgz']:
+        raise ValueError('`t1w` must be a .nii, .nii.gz, or .mgz file, but is'
+                         ' {}'.format(ext))
+    elif ext == '.mgz':
+        t1_mgz = nib.load(t1w)
+        tmp_dir = _TempDir()
+        t1_nifti_path = op.join(tmp_dir, 't1_nifti.nii.gz')
+        nib.save(t1_mgz, t1_nifti_path)
+        t1w = t1_nifti_path  # we need to use the converted file to save BIDS
+        ext = '.nii.gz'  # also update the extension
+    else:  # this is NIfTI
+        t1_nifti = nib.load(t1w)
+        tmp_dir = _TempDir()
+        t1_mgz_path = op.join(tmp_dir, 't1_mgz.mgz')
+        nib.save(t1_nifti, t1_mgz_path)
```

Comment:
This should not be necessary. I think it needs an issue in `nibabel`. @agramfort thoughts?

Reply:
indeed no need to write on disk. I just hacked something:


```
import numpy as np
import nibabel as nib

t1 = nib.load('/Users/alex/mne_data/MNE-sample-data/subjects/sample/mri/T1.nii')


def get_vox2ras(t1):
    return t1.affine


def get_vox2ras_tkr(t1):
    '''Get the vox2ras-tkr transform. Inspired
    by get_vox2ras_tkr in
    '''
    ds = t1.header._structarr['pixdim'][1:4]
    ns = t1.header._structarr['dim'][1:4] * ds / 2.0
    v2rtkr = np.array([[-ds[0], 0, 0, ns[0]],
                       [0, 0, ds[2], -ns[2]],
                       [0, -ds[1], 0, ns[1]],
                       [0, 0, 0, 1]], dtype=np.float32)
    return v2rtkr


nib.save(t1, 'try.mgz')
t1_mgz = nib.load('try.mgz')

np.testing.assert_array_equal(get_vox2ras(t1), t1_mgz.header.get_vox2ras())
np.testing.assert_array_equal(get_vox2ras_tkr(t1), t1_mgz.header.get_vox2ras_tkr())
```

Reply:
there has also been an answer on my neurostars question: https://neurostars.org/t/get-voxel-to-ras-transformation-from-nifti-file/4549/2?u=sappelhoff